app/src/main/java/woowacourse/shopping/data/repository/CartRepositoryImpl.kt

```diff
@@ -0,0 +1,74 @@
+package woowacourse.shopping.data.repository
+
+import woowacourse.shopping.data.dao.CartDao
+import woowacourse.shopping.data.dao.ProductDao
+import woowacourse.shopping.data.entity.CartEntity
+import woowacourse.shopping.data.entity.toDomain
+import woowacourse.shopping.domain.model.CartProduct
+import woowacourse.shopping.domain.model.Product
+import android.os.Handler
+import android.os.Looper
+
+
+class CartRepositoryImpl(
+    private val cartDao: CartDao,
+    private val productDao: ProductDao
+) : CartRepository {
+
+    private val loadItemCount = 5
+    private val mainHandler = Handler(Looper.getMainLooper())
+
+    override fun fetchCartProducts(page: Int, callback: (List<CartProduct>) -> Unit) {
+        Thread {
+            val offset = (page - 1) * loadItemCount
+            val result = cartDao.fetchPagedCart(loadItemCount, offset)
+                .mapNotNull { it.toDomain(productDao) }
+
+            mainHandler.post {
+                callback(result)
+            }
+        }.start()
+    }
```

Comment:
```suggestion
        thread {
            val offset = (page - 1) * loadItemCount
            val result = cartDao.fetchPagedCart(loadItemCount, offset)
                .mapNotNull { it.toDomain(productDao) }
            callback(result)
        }
```

Handler 없이 이런 형태로도 가능했을 것 같은데 Handler의 사용 이유가 궁금하네요!

Reply:
아무래도 페이지UI를 직접적으로 건드는 것이다 보니 메인 쓰레드를 강제하기 위해 사용했습니다..!